config/config.go

```diff
@@ -462,18 +429,48 @@ func parseConfigStringAsTerragruntConfig(configString string, resolveContext *re
 }
 
 // parseHcl uses the HCL2 parser to parse the given string into the struct specified by out.
-func parseHcl(hcl string, filename string, out interface{}, resolveContext *resolveContext) (err error) {
+// If the supplied data is in another format, it converts it to json to use HCL2 ParseJSON instea
+func parseHcl(content string, filename string, out interface{}, resolveContext *resolveContext) (err error) {
 	// The HCL2 parser and especially cty conversions will panic in many types of errors, so we have to recover from
 	// those panics here and convert them to normal errors
-	defer func() {
-		if recovered := recover(); recovered != nil {
-			err = errors.WithStackTrace(panicWhileParsingConfig{RecoveredValue: recovered, ConfigFile: filename})
-		}
-	}()
+	defer errors.Recover(func(cause error) {
+		err = errors.WithStackTrace(panicWhileParsingConfig{RecoveredValue: cause, ConfigFile: filename})
+	})
 
 	parser := hclparse.NewParser()
+	var file *hcl.File
+	var parseDiagnostics hcl.Diagnostics
+
+	convert := func(convert func([]byte, interface{}) error) error {
+		var data map[string]interface{}
+		if err := convert([]byte(content), &data); err != nil {
+			return err
+		}
+		content, err := json.Marshal(data)
+		if err != nil {
+			panic(err)
+		}
+		// It is important to add an extension to the filename since hcl parsers maintain a cache of already seen files
+		file, parseDiagnostics = parser.ParseJSON(content, filename+"retry")
+		return nil
+	}
+
+	switch filepath.Ext(filename) {
+	case ".json":
+		file, parseDiagnostics = parser.ParseJSON([]byte(content), filename)
+	case ".hcl", ".tfvars":
+		file, parseDiagnostics = parser.ParseHCL([]byte(content), filename)
+	case ".yml", ".yaml":
```

Comment:
Instead of hardcoding the extensions here, in `util/terraform.go` also in `options/constants.go`, maybe define constants for these extensions in options/constants.go and refer to these constants everywhere?  To make the code more maintainable since we support so many different extensions.

Reply:
That's an idea, but in that specific case, it is not related to supported extensions, but really to the standard extensions recognized by the industry. Replacing them by constant will not add readability. I am open to discuss it.